qubesadmin/app.py

```diff
@@ -216,10 +216,8 @@ def get_label(self, label):
             pass
 
         # then search for index
-        if label.isdigit():
-            for i in self.labels:
-                if i.index == int(label):
-                    return i
+        if type(label) == int or label.isdigit():
+            return self.labels.values()[int(label)]
```

Comment:
Label.index is _not_ the same as index in what `values()` returns - dict-like objects in Python generally don't have stable ordering. Just adding `isinstance(label, int)` (not `type(...)`) to the previous version would be enough.

Reply:
Ah! And there I was wondering why this code is so non-pythonian. Turns out, I completely misunderstood the problem. So I kept the ability to accept `int`-type arguments and corrected the for-loop. New commit added.